RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc

```diff
@@ -10,13 +10,26 @@
 
 using namespace ticl;
 
-PatternRecognitionbyCA::PatternRecognitionbyCA(const edm::ParameterSet &conf) : PatternRecognitionAlgoBase(conf) {
+PatternRecognitionbyCA::PatternRecognitionbyCA(const edm::ParameterSet &conf, const CacheBase *cache)
+    : PatternRecognitionAlgoBase(conf, cache), eidSession_(nullptr) {
   theGraph_ = std::make_unique<HGCGraph>();
   min_cos_theta_ = conf.getParameter<double>("min_cos_theta");
   min_cos_pointing_ = conf.getParameter<double>("min_cos_pointing");
   missing_layers_ = conf.getParameter<int>("missing_layers");
   min_clusters_per_ntuplet_ = conf.getParameter<int>("min_clusters_per_ntuplet");
   max_delta_time_ = conf.getParameter<double>("max_delta_time");
+  eidInputName_ = conf.getParameter<std::string>("eid_input_name");
```

Comment:
all configurable should be declared ```const``` and initialized as
```theGraph_(std::make_unique<HGCGraph>()),``` etc

Reply:
Please make const all all configurable data members

Reply:
✔︎